src/java/org/apache/cassandra/index/sai/disk/v6/TermsDistribution.java

```diff
@@ -80,19 +84,40 @@ private TermsDistribution(AbstractType<?> termType, List<Bucket> histogram, Navi
         this.termType = termType;
         this.histogram = histogram;
         this.mostFrequentTerms = mostFrequentTerms;
+        // TODO do we need to consider the version here? This is a hack that seems to have made numbers work (even
+        // the normal ones that were not truncated), but I'm not sure it works for text and other columns.
+        // The only way to know is to add tests. What is most odd is that this seems necessary for all numeric
+        // queries to work. On one hand, that makes sense because of how numeric queries encode their search terms
+        // and then compare within that index, but it's still a bit confusing.
+        // TODO add test coverage for strings.
+        this.termComparator = termType instanceof NumberType ? ByteComparable::compare : termType;
 
         this.numRows = histogram.isEmpty() ? 0 : histogram.get(histogram.size() - 1).cumulativeRowCount;
         this.numPoints = histogram.isEmpty() ? 0 : histogram.get(histogram.size() - 1).cumulativePointCount;
         this.minTerm = histogram.isEmpty() ? null : histogram.get(0).term;
         this.maxTerm = histogram.isEmpty() ? null : histogram.get(histogram.size() - 1).term;
     }
 
+
+    private ByteBuffer getComparableBytes(ByteBuffer term)
+    {
+        if (term == null)
+            return null;
+        var bytes = TypeUtil.asComparableBytes(term, termType).asByteComparableArray(TypeUtil.BYTE_COMPARABLE_VERSION);
+        return ByteBuffer.wrap(bytes);
+    }
+
     /**
      * Estimates the number of values equal to the given term.
      *
      * @param term term encoded as byte-comparable
      */
     public long estimateNumRowsMatchingExact(ByteBuffer term)
+    {
+        return estimateNumRowsMatchingExactEncoded(getComparableBytes(term));
```

Comment:
ok, got it; so query bounds encoded are not byte comparable. Can we move it up to the caller then?

Reply:
Right, and we could for now, but if we ever version the code, it'll get pushed down into the method since we'll likely want the raw term to get passed in and then we'll deal with encoding here (that is essentially what we do for index queries already, see the `BKDQueries#bkdQueryFrom` method for details)

Reply:
The raw term is passed to `SegmentMetadata` and encoded there. This is similar to how IndexSearchers work. Therefore `TermsDistribution` works on `ByteComparables` everywhere now.